tools/imas/imas_wrapper.py

```diff
@@ -0,0 +1,542 @@
+		else:
+			j_tor = equilibrium.time_slice[0].profiles_1d.j_tor
+			if log:
+				logging.info('The toroidal current density from equilibrium IDS loaded successfully!\n')
+
+	if equilibrium.time_slice[0].global_quantities.ip == 0:
+		logging.warning('The plasma current from equilibrium IDS seems to be empty!\n')
+		raise Exception('Error in loading equilibrium data. Please check the input data!')
+	else:
+		Ip = equilibrium.time_slice[0].global_quantities.ip
+		if log:
+			logging.info('The plasma current from equilibrium IDS loaded successfully!\n')
+	
+	# get the 2D psi data from equilibrium, and interpolate the core_profiles data onto this grid	
+	# get the 2D grid
+	if len(equilibrium.time_slice[0].profiles_2d[0].r) == 0:
+		logging.warning('The 2D minor radius from equilibrium IDS seems to be empty!\n')
+		raise Exception('Error in loading equilibrium data. Please check the input data!')
+	else:
+		r = equilibrium.time_slice[0].profiles_2d[0].r
+		if log:
+			logging.info('The 2D minor radius from equilibrium IDS loaded successfully!\n')
+
+	if len(equilibrium.time_slice[0].profiles_2d[0].z) == 0:
+		logging.warning('The 2D height coordinate from equilibrium IDS seems to be empty!\n')
+		raise Exception('Error in loading equilibrium data. Please check the input data!')
+	else:
+		z = equilibrium.time_slice[0].profiles_2d[0].z
+		if log:
+			logging.info('The 2D height coordinate from equilibrium IDS loaded successfully!\n')
+	
+	# find the z = 0 column
+	midplane_index = np.where(z==0)[1][0]
+	
+	# get the psi coordinates on the midplane
+	if len(equilibrium.time_slice[0].profiles_2d[0].psi[:, midplane_index]) == 0:
+		logging.warning('The 2D poloidal flux from equilibrium IDS seems to be empty!\n')
+		raise Exception('Error in loading equilibrium data. Please check the input data!')
+	else:
+		eq_psi = equilibrium.time_slice[0].profiles_2d[0].psi[:, midplane_index]
+		if log:
+			logging.info('The 2D poloidal flux from equilibrium IDS loaded successfully!\n')	
+
+	# limit the psi coordinates to the outer midplane
+	# get the limiting psi values
+	magnetic_axis_psi = max(eq_psi)			# psi value on the magnetic axis
+	
+	# get the indices fot the limits in psi_eq
+	# get the magnetic axis index
+	magnetic_axis_psi_index = np.where(eq_psi == max(eq_psi))[0][0]
+	eq_psi = eq_psi[magnetic_axis_psi_index:]
+	
+	#normalize eq_psi
+	eq_psi = eq_psi - eq_psi[0]	
+
+	# load the coreprof psi
+	if len(coreprof.profiles_1d[0].grid.psi) == 0:
+		logging.warning('The poloidal flux from core profiles IDS seems to be empty!\n')
+		raise Exception('Error in loading core profiles data. Please check the input data!')
+	else:
+		cp_psi = coreprof.profiles_1d[0].grid.psi
+		if log:
+			logging.info('The poloidal flux from core profiles IDS loaded successfully!\n')
+	
+	# interpolate the coreprof rho from the coreprof psi to the equilibrium psi
+	interp_radius = CubicSpline(-cp_psi, cp_radius)
```

Comment:
Does IMAS enforce a particular sign convention for poloidal flux? Or why do you negate ``psi`` here?

Reply:
I negate psi because the input psi function is a decreasing function and the interpolation requires a monotonically increasing one. I was looking how to solve this issue more elegantly, but this was what worked in the end.
Is there a better solution you can recommend?

Reply:
Hmm... The sign of d(psi)/dr is typically quite important as it is chosen to follow a certain convention. Here it seems that you are interpolating from psi -> r, so the convention might not be too important, but shouldn't this just yield the wrong result? I think the desired solution is rather to sort the two arrays, i.e.
```
p = cp_psi.argsort()
cp_psi = cp_psi[p]
cp_radius = cp_radius[p]
```

Reply:
I am not interpolating from psi to r. I have a psi coordinate as a function of r in the Equilibrium IDS and the same in Core profiles. I get the midplane psi values from the 2D equilibrium, and interpolate the rho coordinates in Coreprof from the coreprof psi (which is not necessarily the midplane) to the equilibrium mid plane psi. The resulting rho is used as the radial coordinate in DREAM. Psi is later not used in the code, it is only used as an interpolation coordinate.

Reply:
Sorry, my point is that there are different conventions for what the sign of ``psi`` should be, and although for the particular dataset you have tested with now, it is negative and decreases towards the edge, that might not always be the case. If you just negate ``psi``, then it might work now for TCV, but maybe not for a different machine. So if the problem is that ``psi`` is expected to be monotonically increasing by the interpolation routine, then my suggestion would directly address this problem, regardless of the sign of ``psi``.

Reply:
I added this, as requested. I was not able to fully test it however, as the DREAM gave an error I was not able to fix. It was complaining about interpolation, once about x not monothonically increasing, once about x being out of range. I am not sure which variables were the problems, as it seemed a general error message from the interpolation method in c, without specifying where DREAM ran into problems. I will try to figure it out, but so far I was not successful.

Reply:
That sounds like a GSL error, coming from the GSL interpolation routines. Basically, it means that some input data, which we interpolate in, has an x vector which is not monotonically increasing.

Reply:
Fixed them, everything works now.